CMakeLists.txt

```diff
@@ -27,6 +27,24 @@ option(run_reals_check "set run_reals_check to ON to run reals check (default is
 option(use_cppunittest "set use_cppunittest to ON to build CppUnitTest tests on Windows (default is OFF)" OFF)
 option(run_traceability "run traceability tool (default is ON)" ON)
 
+if((NOT "${CMAKE_BUILD_TYPE}" STREQUAL "RelWithDebInfo") AND (NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Debug"))
+    message(FATAL_ERROR "CMAKE_BUILD_TYPE must be specified and be precisely one of Debug or RelWithDebInfo")
+endif()
```

Comment:
why is this needed?

Asking because I imagine this cmakelists.txt would "go down" in hierarchy all the way to c-pal/c-build-tools and it would fail there.

Reply:
this = all the yellow lines here.

Reply:
I removed them. I was torn between explicitly enforcing this on each repo or relying on c-pal (in the future c-build-tools I guess) to do it. Because this should migrate to c-build-tools methinks.